src/lib/repositories_v2/impl/inventory_ingredient_repository_impl.py

```diff
@@ -0,0 +1,95 @@
+
+    def add(self, inventory_ingredient):
+        inventory_ingredient.created_date = datetime.now()
+        inventory_ingredient.updated_by = inventory_ingredient.created_by
+        inventory_ingredient.updated_date = inventory_ingredient.created_date
+        self.session.add(inventory_ingredient)
+        self.session.commit()
+
+    def get_by_id(self, inventory_ingredient_id):
+        return (
+            self.session.query(InventoryIngredient)
+            .filter(InventoryIngredient.id == inventory_ingredient_id)
+            .filter(InventoryIngredient.entity_status == Status.ACTIVE.value)
+            .first()
+        )
+
+    def get_all(self):
+        inventory_ingredients = self.session.query(InventoryIngredient).filter(
+            InventoryIngredient.entity_status == Status.ACTIVE.value
+        )
+        return list(inventory_ingredients)
+
+    def delete_by_id(self, inventory_ingredient_id, inventory_ingredient):
+        self.session.query(InventoryIngredient).filter(
+            InventoryIngredient.id == inventory_ingredient_id
+        ).update(
+            {
+                InventoryIngredient.entity_status: Status.DELETED.value,
+                InventoryIngredient.updated_date: datetime.now(),
+                InventoryIngredient.updated_by: inventory_ingredient.updated_by,
+            }
+        )
+        self.session.commit()
+
+    def update_by_id(self, inventory_ingredient_id, inventory_ingredient):
+        inventory_ingredient_to_be_updated = (
+            self.session.query(InventoryIngredient)
+            .filter(InventoryIngredient.id == inventory_ingredient_id)
+            .first()
+        )
+        inventory_ingredient_to_be_updated.inventory_id = (
+            inventory_ingredient.inventory_id
+            or inventory_ingredient_to_be_updated.inventory_id
+        )
+        inventory_ingredient_to_be_updated.ingredient_id = (
+            inventory_ingredient.ingredient_id
+            or inventory_ingredient_to_be_updated.ingredient_id
+        )
+        inventory_ingredient_to_be_updated.quantity = (
+            inventory_ingredient.quantity or inventory_ingredient_to_be_updated.quantity
+        )
+
+        inventory_ingredient_to_be_updated.updated_date = datetime.now()
+        inventory_ingredient_to_be_updated.updated_by = inventory_ingredient.updated_by
+        self.session.add(inventory_ingredient_to_be_updated)
+        self.session.commit()
+
+    def get_by_ingredient_id(self, ingredient_id):
+        inventory_ingredients = (
+            self.session.query(InventoryIngredient)
+            .filter(InventoryIngredient.entity_status == Status.ACTIVE.value)
+            .filter(InventoryIngredient.ingredient_id == ingredient_id)
+        )
+        return list(inventory_ingredients)
+
+    def get_final_product_qty_by_product_ids(self, product_ids):
+        reduce_products_qty_array_to_final_products_qty_map = (
```

Comment:
Do you really need to do this? I feel this can be done in a query

Reply:
done.